pycalphad/mapping/strategy/ternary_strategy.py

```diff
@@ -153,36 +155,62 @@ def initialize(self):
+
+                # Iterate through the zpf line until we find a point with the same number of phases as the
+                # zpf line itself. This seems redundant, but it is because the first point of the zpf line
+                # is likely going to be a node or a node parent, where the number of phases may differ by 1
                 p_index = 0
                 while len(zpf_line.points[p_index].stable_phases) != num_phases:
                     p_index += 1
 
+                # If we do find a good point on the zpf line, then we can process it to be a starting point
                 if len(zpf_line.points[p_index].stable_phases) == num_phases:
                     new_point = zpf_line.points[p_index]
                     if self.all_vars[-1] in new_point.global_conditions:
                         del new_point.global_conditions[self.all_vars[-1]]
                         new_point.global_conditions[self.axis_vars[0]] = new_point.get_property(self.axis_vars[0])
                     _log.info(f"Adding node {new_point.fixed_phases}, {new_point.free_phases}, {new_point.global_conditions}")
 
+                    # New point will be make with all composition sets as free, since we set
+                    # which CS to fix when finding an exit
                     free_point = map_utils._generate_point_with_free_cs(new_point)
 
-                    if self._check_full_global_equilibrium(free_point, add_global_point_if_false = True):
+                    cs_result = zeq._find_global_min_cs(free_point, system_info=self.system_info, pdens=self.GLOBAL_MIN_PDENS, tol=self.GLOBAL_MIN_TOL, num_candidates=self.GLOBAL_MIN_NUM_CANDIDATES)
+                    if cs_result is None:
                         new_node = self._create_node_from_point(free_point, None, None, None)
                         self.node_queue.add_node(new_node)
 
+    def _validate_custom_starting_point(self, point: Point, direction: Direction):
+        """
+        Modifies exit hint and direction based off number of composition sets
+            Single phase -> cannot be added
+            Two phase    -> point is exit, direction stays the same as input
+            Three phase  -> normal exit finding strategy, no direction needed
+
+        This is the same as BinaryStrategy
+        """
+        if len(point.stable_composition_sets) <= 1:
+            return None, None, "Single phase detected"
+        elif len(point.stable_composition_sets) == 2:
+            return ExitHint.POINT_IS_EXIT, direction, None
+        elif len(point.stable_composition_sets) == 3:
+            return ExitHint.NORMAL, None, None
+        else:
+            return None, None, "More than three phases detected"
+
     def _find_exits_from_node(self, node: Node):
         """
-        A node on for a binary system has three exits, which are combinations of 2 CS in the node
+        A node on for a ternary system has three exits, which are combinations of 2 CS in the node
```

Comment:
"A node on"

Reply:
Replaced "A node on for a" to "A node in". One may question how little coffee I had when I wrote this.